### PR TITLE
Less verbose integration test output

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

#2119 

# Overview

- [x] More quiet stdout per default when running integration tests
- [x] Log full wallet, node and test-setup logs to `$TESTS_LOGDIR`, if set.
- [x] Upload `$TESTS_LOGDIR/*.log` as Buildkite artefacts

# Comments

- Regardless of `$TESTS_LOGDIR` setting, logs are still written to the temporary cluster directory as before.
- We can make hydra and GitHub Actions upload logs in a future PR. We might also be fine without it.

Skärmavbild 2020-09-10 kl  17 57 58

### CODE REVIEW

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

(this doesn't make sense 🤕 )

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

(It won't run the full integration tests now, so should be quick)

**Comment by Anviking** [comment]
It seems the integration test themselves don't see the `TESTS_LOGDIR` env var as set, despite it being set in `rebuild.hs`

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
If I have a small script
```haskell
cat debug.sh
#! /bin/bash
echo $TESTS_LOGDIR
```
and modify `rebuild.hs` to first read the `TESTS_LOGDIR` env var itself, and then run the `debug.sh` script,

and I run
```
export TESTS_LOGDIR=foo
nix-build .buildkite/default.nix -o sr
./sr/bin/rebuild
```
I see
```
rebuild.hs: TESTS_LOGDIR=Just "foo"
foo
ExitSuccess
```
but for some reason, when `stack test` is run instead of the script, the `TESTS_LOGDIR` variable isn't passed a long. I haven't managed to reproduce the latter locally yet though.

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Comment by rvl** [comment]
I fixed the error on Hydra, and cleaned up stack.yaml.

bors try

**Comment by Anviking** [comment]
bors try

`nix pure: false` didn't work for me previously, checking with added debug logs

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

**Comment by Anviking** [comment]
bors try

File: lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs [hunk 1]
```diff
@@ -777,19 +796,37 @@ genConfig
...
+                , scMinSev = sev
+                , scMaxSev = Critical
+                , scPrivacy = ScPublic
+                , scRotation = Nothing
+                }
+
+    let scribes = catMaybes
+            [ Just $ fileScribe ("cardano-node.log", severity)
+            , fileScribe . first T.pack <$> mExtraLogFile
+            ]
```
Comment (KtorZ) [inline lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs:821]: What's the use for two log files here?

File: lib/shelley/test/integration/Main.hs [hunk 2]
```diff
@@ -242,26 +246,29 @@ specWithServer (tr, tracers) = aroundAll withContext . after tearDown
...
                 dir
+                extraLogDir
                 onByron
                 (afterFork dir)
                 (onClusterStart action dir dbDecorator)
 
     tr' = contramap MsgCluster tr
     onByron _ = pure ()
     afterFork dir _ = do
+        traceWith tr MsgSettingUpFacuet
```
Comment (KtorZ) [inline lib/shelley/test/integration/Main.hs]: *Facuet

File: lib/shelley/test/integration/Main.hs [hunk 3]
```diff
@@ -348,13 +370,19 @@ bracketTracer' tr name = bracketTracer (contramap (MsgBracket name) tr)
...
-    minSeverityFromEnv Error "CARDANO_NODE_TRACING_MIN_SEVERITY"
+    minSeverityFromEnv Info "CARDANO_NODE_TRACING_MIN_SEVERITY"
 
--- Allow configuring integration tests and wallet log level with
+-- Allow configuring wallet log level with
 -- @CARDANO_WALLET_TRACING_MIN_SEVERITY@ environment variable.
 walletMinSeverityFromEnv :: IO Severity
 walletMinSeverityFromEnv =
-    minSeverityFromEnv Info "CARDANO_WALLET_TRACING_MIN_SEVERITY"
+    minSeverityFromEnv Critical "CARDANO_WALLET_TRACING_MIN_SEVERITY"
```
Comment (KtorZ) [inline lib/shelley/test/integration/Main.hs:379]: Is it really sound? Sure we will have less logs but we still needs those logs.

**Comment by KtorZ** [comment]
That doesn't quite work locally for me. 

```
Progress 7/8: cardano-wallet--- TESTS_LOGDIR env:
integration: user error (Pattern match failure in do expression at test/integration/Main.hs:146:5-19)
```

**Comment by KtorZ** [comment]
bors merge

**Review by KtorZ — APPROVED** [review]

**Comment by KtorZ** [comment]
bors merge

**Comment by KtorZ** [comment]
bors merge

File: lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs (hunk 1 above)
Comment (Anviking) [reply lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs:821]: It seemed drastic/weird to remove the existing node logs from the temporary directory.

Without setting additional environment variables, we can inspect the dbs, and the logs, in that directory, while the tests are running.

Thought it is also a bit weird to duplicate the logs 🤷‍♂️

File: lib/shelley/test/integration/Main.hs (hunk 3 above)
Comment (Anviking) [reply lib/shelley/test/integration/Main.hs:379]: This is the severity used for `Stdout`.

**Pull request merged**